include/lbann/base.hpp

```diff
@@ -126,12 +130,33 @@ matrix_format data_layout_to_matrix_format(data_layout layout);
 /// Neural network execution mode
 enum class execution_mode {training, validation, testing, prediction, invalid};
 std::string to_string(execution_mode m);
+typedef Iterator<execution_mode, execution_mode::training, execution_mode::invalid> execution_mode_iterator;
```

Comment:
Prefer `using` statements.
```suggestion
using execution_mode_iterator = Iterator<execution_mode, execution_mode::training, execution_mode::invalid>;
```